### DISCUSSION TITLE
can't connect db in container (node 16)

### DISCUSSION THREAD

**Discussion by Atsuyoshi-Funahashi** [prisma/prisma#14970] [Q&A]
Hi. I am currently creating an application in Next.js and using mysql for the DB.
I am building the environment with docker. Originally I was using node version 17.6.0 and it was working fine.

## Problem
I have lowered my node version to 16.13.0 in order to install NextAuth.
After that, when I launch the container with `docker compose up --build`, the following error occurs during migration.

```
# error
Error: P1001: Can't reach database server at `mysql`:`3306`
Please make sure your database server is running at `mysql`:`3306`
```
Since this did not occur when node version was 17, I would like to know how to solve this problem.

## Version
```
node  16.13.0
prisma 4.2.1
mysql 5.7
MacOS Monterey 12.4 (M1)
```

## reproduction method
```
1  create next app
2 install and init prisma
3 Copy-paste the file below.
4 docker compose up --build
```

## File
Dockerfile, docker-compose.yml, run.sh, and .env are as follows
```
## Dockerfile
FROM node:16.13.0

RUN mkdir /test
WORKDIR /test
COPY . /test
RUN yarn add create-next-app
COPY run.sh /usr/bin/run
RUN chmod +x /usr/bin/run
EXPOSE 3000

CMD ["run"]
```

```
## docker-compose.yml
version: '3'
services:
  db:
    image: mysql:5
    platform: linux/x86_64
    container_name: mysql
    ports:
      - '3306:3306'
    environment:
      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - ./tmp/db:/var/lib/mysql
  app:
    build: .
    volumes:
      - .:/test
    ports:
      - '3000:3000'
    restart: always
    depends_on:
      - db

```

This is schema.prisma.
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model users {
  id               Int       @id @default(autoincrement())
  email            String    @unique
 # other value
}

```
This is run.sh. Please set under the project root.
```
# run.sh
#!/bin/sh

bash -c "sleep 30 && yarn -s run prisma migrate dev --name init && yarn install && yarn dev "
```

This is .env
```
# .env
MYSQL_HOST = 'mysql'
MYSQL_DATABASE = 'database_name'
# MYSQL_USER = 'root'
MYSQL_ROOT_PASSWORD = ''
DATABASE_URL="mysql://root:@mysql:3306/database_name"

```

**Comment by nurul3101** [comment]
Hey @Atsuyoshi-Funahashi 👋 
As changing the node version caused this issue for you, I would suggest you try with the LTS version of Node 16 which v16.17.0.
Could you use this LTS node version and check if the issue is still happening?

**Reply by Atsuyoshi-Funahashi** [reply]
Hi, thank you for your comment.
Conclusively, I have found a solution.

I already have checked in version v16.17.0, but it doesn't works.
I found that my intel Mac didn't cause this problem, so it might be a problem about M1 Mac.

So I changed the platform of app as good as mysql image like this. This worked!

```
## docker-compose.yml
version: '3'
services:
  db:
    image: mysql:5
    platform: linux/x86_64
    container_name: mysql
    ports:
      - '3306:3306'
    environment:
      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - ./tmp/db:/var/lib/mysql
  app:
    build: .
    platform: linux/x86_64
    volumes:
      - .:/test
    ports:
      - '3000:3000'
    restart: always
    depends_on:
      - db

```

**Reply by nurul3101** [reply]
Glad to hear that! 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
